### PR TITLE
Conforming Base32 to RFC 4648 and re-enabling it

### LINKED ISSUE #230: Base32 encoding does not follow RFC 4648 spec
https://tools.ietf.org/html/rfc4648#section-6

The base32 encoding alphabet specified by RFC 4648 is not used here, or even offered as an option, but it should be the default.

As specified in the document linked above, the standard base32 alphabet should be `ABCDEFGHIJKLMNOPQRSTUVWXYZ234567`.

It'd be nice to offer the base32hex alphabet as well: `0123456789ABCDEFGHIJKLMNOPQRSTUV`

### PR DESCRIPTION
**Pull request by robyoder**
1. Changed default `base32` encoding to use alphabet specified by [RFC 4648](http://tools.ietf.org/html/rfc4648#section-6)
2. Added proper `=` padding
3. Added `base32hex` encoding, also documented in RFC 4648. This is the same as the previous default except using uppercase letters. (Case doesn't matter, but spec uses uppercase. User can downcase if desired.)
4. Re-enabled codecBase32 in sjcl build (disabled here: 11a673d)

Fixes #230

### CODE REVIEW

File: config.mk [hunk 1]
```diff
@@ -1,2 +1,2 @@
-SOURCES= core/sjcl.js core/aes.js core/bitArray.js core/codecString.js core/codecHex.js core/codecArrayBuffer.js core/codecBase64.js core/sha256.js core/ccm.js core/ccmArrayBuffer.js core/ocb2.js core/gcm.js core/hmac.js core/pbkdf2.js core/random.js core/convenience.js
```
Comment (robyoder) [inline config.mk]: This was just generated by `make distclean`, must've been removed previously without rebuilding?

**Comment by Nilos** [comment]
Am I correct that, when decoding legacy encoded base32 one would have to call the base32hex function?

**Comment by robyoder** [comment]
@Nilos yes, that is correct. I just added a fallback in `toBits` that would try `base32hex` if an invalid `base32` character is found. But that won't catch strings that are valid as both `base32` and `base32hex`.

The upside, though, is that the base32 codec has not been enabled on master for quite a while, so this is a good time to make this change. It was enabled from [2014/08/20](commit:44d69ff) to 2015/01/25 (11a673d)

**Comment by Nilos** [comment]
Yes, I would at least like to add a note about this in the Readme.
Actually I am affected by this in my own private project.

**Comment by robyoder** [comment]
@Nilos Ok, I can add a note. Can I bump the version number to 1.0.4 to make the distinction?

**Comment by Nilos** [comment]
Sure go ahead!

**Comment by robyoder** [comment]
@Nilos done

**Pull request merged**